src/ServiceControl/MessageFailures/FailedMessagesFeature.cs

```diff
@@ -54,12 +54,14 @@ public override void Enrich(IReadOnlyDictionary<string, string> headers, IDictio
                 if (isOldRetry)
                 {
                     bus.Publish<MessageFailureResolvedByRetry>(m => m.FailedMessageId = headers.UniqueId());
+                    return;
                 }
 
-                if (isNewRetry)
+                bus.CurrentMessageContext.Headers.Add("ServiceControl.Retry.UniqueMessageId", headers.UniqueId());
```

Comment:
Does this work? I believe that `CurrentMessageContext.Headers` refers to the headers on the message being processed, not the headers on outgoing messages.

This type is OK to add a property to because although it is in the Contracts namespace it is not actually exposed externally. The only types we don't want to make changes to are the ones found in the [`ServiceControl.Contracts` assembly](https://github.com/Particular/ServiceControl.Contracts/tree/master/src/ServiceControl.Contracts).